string[] labels
bool moveHead
---
string[] labelList
int32[] labelFound
---
#feedback
int32 feedback
